deeposlandia/datasets/aerial.py

```diff
@@ -0,0 +1,144 @@
+    by INRIA
+
+    The dataset is freely available at:
+    https://project.inria.fr/aerialimagelabeling/files/
+
+    It is composed of 180 training images and 180 testing images of size
+    5000*5000. There are ground-truth labels only for the former set.
+
+    Attributes
+    ----------
+    tile_size : int
+        Size of the tiles into which each raw images is decomposed during
+    dataset population (height=width)
+
+    """
+
+    def __init__(self, tile_size):
+        """ Class constructor ; instanciates a AerialDataset as a standard
+        Dataset which is completed by a glossary file that describes the
+        dataset labels and images
+
+        """
+        self.tile_size = tile_size
+        img_size = utils.get_image_size_from_tile(self.tile_size)
+        super().__init__(img_size)
+        self.add_label(label_id=0, label_name="background",
+                       color=0, is_evaluate=True)
+        self.add_label(label_id=1, label_name="building",
+                       color=255, is_evaluate=True)
+
+    def _preprocess(self, image_filename, output_dir, labelling):
+        """Resize/crop then save the training & label images
+
+        Parameters
+        ----------
+        image_filename : str
+            Full path towards the image on the disk
+        datadir : str
+            Output path where preprocessed image must be saved
+
+        Returns
+        -------
+        dict
+            Key/values with the filenames and label ids
+        """
+        img_in = Image.open(image_filename)
+        raw_img_size = img_in.size[0]
+        result_dicts = []
+        # crop tile_size*tile_size tiles into 5000*5000 raw images
+        buffer_tiles = []
+        for x in range(0, raw_img_size, self.tile_size):
+            for y in range(0, raw_img_size, self.tile_size):
+                tile = img_in.crop((x, y,
+                                    x + self.tile_size, y + self.tile_size))
+                tile = utils.resize_image(tile, self.image_size)
+                img_id = int((raw_img_size / self.tile_size
+                              * x / self.tile_size
+                              + y / self.tile_size))
+                basename_decomp = os.path.splitext(
+                    os.path.basename(image_filename))
+                new_in_filename = (basename_decomp[0] + '_' +
+                                   str(img_id) + basename_decomp[1])
+                new_in_path = os.path.join(output_dir, 'images', new_in_filename)
+                tile.save(new_in_path.replace(".tif", ".png"))
+                result_dicts.append({"raw_filename": image_filename,
+                                     "image_filename": new_in_path})
+
+        if labelling:
+            label_filename = image_filename.replace("images/", "gt/")
+            img_out = Image.open(label_filename) 
+            buffer_tiles = []
+            for x in range(0, raw_img_size, self.tile_size):
```

Comment:
this double loop looks like with the previous one. Don't you think it's possible to refactor this ?

Reply:
It certainly is. Could we improve that particular point in a subsequent PR?

Reply:
I have just created a new issue to address this problem. I propose to close the discussion in this PR scope.